Add updating Jenkins section in User's Handbook

Vandit1604 opened the pull request:
Windows and Linux sections are based on my knowledge and @krisstern helped in macOS section for upgrading jenkins other than that every thing is compiled by me after reading blogs and questions answered on community forum of  jenkins 
feel free to add upgrading instructions for other sections 
this closes #5059

krisstern commented:
Hi @Vandit1604 Try adding a file with your GitHub username as the file name in here for the authors: https://github.com/jenkins-infra/jenkins.io/tree/master/content/_data/authors

Vandit1604 commented:
@krisstern Hi I created this PR #5890 doing what you instructed
thanks for your time

krisstern commented:
Maybe it would be better originally to have the changes you made in PR #5890 to this PR instead. Now you will have to wait until that PR is merged and pull the latest changes to this PR later before every check is passed.

Vandit1604 commented:
I can close that PR #5890  and commit the changes to this PR so this can be merged/reviewed much faster

krisstern commented:
Sure

krisstern commented:
@Vandit1604 It appears there are some bugs to be patched before the checks are passed:
```shell
[2023-01-07T14:30:23.323Z] Failing build due to warnings in log output:

[2023-01-07T14:30:23.323Z] asciidoctor: WARNING: doc/book/updating/_update-wizard.adoc: line 16: section title out of sequence: expected level 2, got level 4

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/download/index.html.haml:156: warning: calling URI.open via Kernel#open is deprecated, call URI.open directly or use URI#open

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/download/index.html.haml:97: warning: calling URI.open via Kernel#open is deprecated, call URI.open directly or use URI#open

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_layouts/corebaseline.html.haml:9: warning: calling URI.open via Kernel#open is deprecated, call URI.open directly or use URI#open

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:11: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:13: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:15: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:16: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.323Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:17: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:20: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:22: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:24: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:26: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:27: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:28: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/_partials/changelog-changes.xml.haml:9: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/security/advisories/rss.xml.haml:40: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/security/advisories/rss.xml.haml:42: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/security/advisories/rss.xml.haml:46: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/security/advisories/rss.xml.haml:48: warning: Using the last argument as keyword parameters is deprecated

[2023-01-07T14:30:23.324Z] /home/jenkins/workspace/Websites_jenkins.io_PR-5888/content/security/advisories/rss.xml.haml:52: warning: Using the last argument as keyword parameters is deprecated
```

krisstern commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,44 @@
+
+
+[[setup-wizard]]
+== Post-Updation setup wizard
```
```suggestion
== Post-update setup wizard
```

krisstern commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,44 @@
+Before you can access Jenkins, there are a few post upgrade steps you'll need
+to perform.
+
+===== Verify the Version update
```
```suggestion
=== Verify the version update
```

krisstern commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,44 @@
+
+The upgraded Jenkins version can be seen at the bottom right corner of the Jenkins UI.
+
+=== Compatibility with the New Version
```
```suggestion
=== Compatibility with the new version
```

krisstern commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,44 @@
+Before you can access Jenkins, there are a few post upgrade steps you'll need
+to perform.
+
+=== Verify the Version update
```
```suggestion
=== Verify the version update
```

krisstern commented:
Currently the images are not showing yet in the preview version at https://deploy-preview-5888--jenkins-io-site-pr.netlify.app/doc/book/updating and some pages like [the one on Windows](https://deploy-preview-5888--jenkins-io-site-pr.netlify.app/doc/book/updating/windows) appear to need some polish up (like with additional description/explanation for each step added). I will let the @jenkins-infra/copy-editors team to review the PR in more depth.

Vandit1604 commented:
I'll also correct the heading size as right now the page looks like a mess

krisstern commented:
For an example of a working link to images:
```
[.boxshadow]
image:tutorials/setup-jenkins-02-unlock-jenkins-page.png[alt="Unlock Jenkins page",width=80%]
```
This is taken from https://raw.githubusercontent.com/jenkins-infra/jenkins.io/master/content/doc/book/installing/windows.adoc

krisstern commented:
> I'll also correct the heading size as right now the page looks like a mess

Sure, sounds like a good idea

Vandit1604 commented:
> For an example of a working link to images:
> 
> ```
> [.boxshadow]
> image:tutorials/setup-jenkins-02-unlock-jenkins-page.png[alt="Unlock Jenkins page",width=80%]
> ```

I have added the images just like that after seeing how images were added in other pages
I'll check it again if the path to the image is broken or some image has the same name

Vandit1604 commented:
Sorry for the commits
I was resolving some conflicts and didn't realise i was committing old changes too

krisstern commented:
You have for the image link now for example: `image::/content/images/updating/check-jenkins-version.png[alt="Checking Version of Updated Jenkins Instance",width=80%]`

I think it should really be like the following with boxshadow to keep the styling consistent:
```shell
[.boxshadow]
image:updating/check-jenkins-version.png[alt="Unlock Jenkins page",width=80%]
```

krisstern commented:
And @Vandit1604 You may want to put your image in here instead: https://github.com/jenkins-infra/jenkins.io/tree/master/content/doc/book/resources

Vandit1604 commented:
> You have for the image link now for example: `image::/content/images/updating/check-jenkins-version.png[alt="Checking Version of Updated Jenkins Instance",width=80%]`
> 
> I think it should really be like the following with boxshadow to keep the styling consistent:
> 
> ```shell
> [.boxshadow]
> image:updating/check-jenkins-version.png[alt="Unlock Jenkins page",width=80%]
> ```

this was how it was before i checked in asciidoctor docs there it is mentioned you can use either (:) or (::) colon but i'll change it to one colon (:)

krisstern commented:
> > You have for the image link now for example: `image::/content/images/updating/check-jenkins-version.png[alt="Checking Version of Updated Jenkins Instance",width=80%]`
> > I think it should really be like the following with boxshadow to keep the styling consistent:
> > ```shell
> > [.boxshadow]
> > image:updating/check-jenkins-version.png[alt="Unlock Jenkins page",width=80%]
> > ```
> 
> this was how it was before i checked in asciidoctor docs there it is mentioned you can use either (:) or (::) colon but i'll change it to one colon (:)

Look I gave you the exact path you are supposed to use, in a folder called `updating` within the path [content/doc/book/resources](https://github.com/jenkins-infra/jenkins.io/tree/master/content/doc/book/resources). Please be careful with reading the instruction. I was not suggesting just a symbol change.

krisstern commented:
@Vandit1604 The paths you are using are currently all broken paths.

Vandit1604 commented:
> Look I gave you the exact path you are supposed to use, in a folder called updating within the path [content/doc/book/resources](https://github.com/jenkins-infra/jenkins.io/tree/master/content/doc/book/resources). Please be careful with reading the instruction. I was not suggesting just a symbol change.

Sorry 
I have committed the correct path now

krisstern commented:
You may want to clean up this PR with the extra commits not intended for this PR.

Vandit1604 commented:
@krisstern Looks good now ?

MarkEWaite reviewed (changes requested):
Thanks so **very much** for creating the section to upgrade Jenkins.

Please review these two videos for more description of the upgrade process:

## How to safely upgrade to a new version

[![How to safely upgrade to a new version](https://img.youtube.com/vi/fEO7dOxHvgU/0.jpg)](https://www.youtube.com/watch?v=fEO7dOxHvgU)

## Upgrading Jenkins Java Version From 8 to 11

[![How to safely upgrade to a new version](https://img.youtube.com/vi/L2Uomz8RWUM/0.jpg)](https://www.youtube.com/watch?v=L2Uomz8RWUM)

It would be very good to embed those videos into one or more of the upgrade pages as we've done with the installation videos for [CentOS / Red Hat / Alma / Rocky](https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos)

krisstern commented on content/doc/book/updating/macos.adoc:
````diff
@@ -0,0 +1,29 @@
+Similarly, to upgrade the Jenkins Weekly on macOS, do the following at the terminal:
+[source]
+```
+brew update && brew install jenkins
````
Yeah, better to use the official recommendation

krisstern commented:
The only concern I have regarding CSS is the overlapping layout here for the Linux upgrading instructions, but it is not fatal, at least for now: 

<img width="848" alt="Screenshot 2023-01-09 at 3 18 47 PM" src="https://user-images.githubusercontent.com/88480540/211257664-697c3167-e842-48e7-922e-76547e558dfe.png">

Vandit1604 commented:
Thanks @MarkEWaite and @krisstern 
I'll do the suggested revisions 
@krisstern Related to the CSS problem can you give me the source css file 
I'll correct that too

krisstern commented:
@Vandit1604 There is actually a quicker way to commit suggested changes on GitHub in batch. Next time, directly do to the `Files Changed` tab and add the changes one-by-one to the same batch, before clicking on a button on the same page to commit everything in bulk. That way would be more efficient (and create less noise for those who are watching this repo too).

krisstern commented:
> Thanks @MarkEWaite and @krisstern I'll do the suggested revisions @krisstern Related to the CSS problem can you give me the source css file I'll correct that too

For the current website build setup we do not have one CSS file for each layout, but I suggest finding out where is best to place the CSS code on your own first, before I make more suggestions. We are currently using Awestruck as the build tool, which is no longer maintained and does not provide ready documentation versioning. One of the GSoC 2022 projects is actully to migrate the jenkins.io website to using a more modern build tool like Antora built on AsciiDoc.

Vandit1604 commented:
>  There is actually a quicker way to commit suggested changes on GitHub in batch. Next time, directly do to the `Files Changed` tab and add the changes one-by-one to the same batch, before clicking on a button on the same page to commit everything in bulk. That way would be more efficient (and create less noise for those who are watching this repo too).

Sorry for the noise i didn't know that we can commit the suggested changes in bulk 
I'll keep that in mind next time

Vandit1604 commented:
> > Thanks @MarkEWaite and @krisstern I'll do the suggested revisions @krisstern Related to the CSS problem can you give me the source css file I'll correct that too
> 
> For the current website build setup we do not have one CSS file for each layout, but I suggest finding out where is best to place the CSS code on your own first, before I make more suggestions. We are currently using Awestruck as the build tool, which is no longer maintained and does not provide ready documentation versioning. One of the GSoC 2022 projects is actully to migrate the jenkins.io website to using a more modern build tool like Antora built on AsciiDoc.

Yeah I know about that Discussion is still going between Hugo and Antora 
But Antora would be the better candidate as Hugo goes more into the security aspects while Antora provides the Documentation based on versions 
Then I'll see what can be done for that CSS problem 
Thanks for the info

krisstern commented:
> Sorry for the noise i didn't know that we can commit the suggested changes in bulk I'll keep that in mind next time

No worries, we all learn by trial-and-error sometimes.

Vandit1604 commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,47 @@
+The Jenkins server is up and running. Let's now verify the version from the Jenkins dashboard:
+
+[.boxshadow]
+image:updating/check-jenkins-version.png[alt="Checking Version of Updated Jenkins Instance",width=80%]
```
Hey @MarkEWaite I have taken the screenshots in the dimensions you said except for the last image as i couldn't able to recreate the warning on the dashboard so removing the image will be the best thing to do for now 
what are your thoughts ?

Vandit1604 commented:
For now i have added some more chapters for upgrading and i'm still working on the table of content problem as due to its css its floating over the content. I have added some more details in some sections if some thing is still not correct feel free to suggest changes
thank you for your time

krisstern commented:
Maybe on the page such as `content/doc/book/updating/index.adoc` try to add the line `:toc: left` such that you will have something like:
```JavaScript
ifdef::backend-html5[]
...
:toc: left
...
endif::[]
```

krisstern commented:
Actually you can add this property to all pages with a TOC

krisstern commented:
This is in the error log for the failed build: 
```
asciidoctor: ERROR: linux.adoc: line 65: level 0 sections can only be used when doctype is book
```

krisstern commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,74 @@
+=== 5. Copy the new jenkins.war you downloaded into your Jenkins installation directory
+This is the upgraded jenkins.war which will update your Jenkins Instance
+
+=== 6.Start Jenkins service
```
```suggestion
=== 6. Start Jenkins service
```

krisstern commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,74 @@
+=== 6.Start Jenkins service
+
+To start the service you need to open the terminal in the directory where your Java WAR File is present and run the following command
+
```
```suggestion
```

Vandit1604 commented:
> This is in the error log for the failed build:
> 
> ```
> asciidoctor: ERROR: linux.adoc: line 65: level 0 sections can only be used when doctype is book
> ```

that is odd as line 65 is inside the codeblock

krisstern commented on content/doc/book/updating/servlet-containers.adoc:
```diff
@@ -0,0 +1,42 @@
+
+== Tomcat
+Use environment variable CATALINA_OPTS to add:
+
```
```suggestion
```

krisstern commented on content/doc/book/updating/other.adoc:
```diff
@@ -0,0 +1,40 @@
+
+When Jenkins is installed from the FreeBSD `pkg` system, it should be upgraded with `pkg`.
+Connect to your FreeBSD machine, get root access and run the commands:
+
```
```suggestion
```

krisstern commented on content/doc/book/updating/macos.adoc:
```diff
@@ -0,0 +1,33 @@
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+endif::[]
+
```
```suggestion
```

krisstern commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,74 @@
+To stop Jenkins Please avoid shutting down the Java process or the Windows service. These are not usual commands. Use those only if your Jenkins is causing problems.
+
+use the following command to stop server
+
```
```suggestion
```

krisstern commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,74 @@
+=== 2. Go to the location where your Jenkins instance is installed
+	
+By Default Jenkins is installed here: 
+
```
```suggestion
```

krisstern commented on content/doc/book/updating/war-file.adoc:
```diff
@@ -0,0 +1,41 @@
+
+include::doc/book/updating/_pre-upgrade-wizard.adoc[]
+
+If your Jenkins instance is run with the command 
```
```suggestion
If your Jenkins instance is run with the command 
[source]
```

krisstern commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,107 @@
+The RPM package contains a post-install script to ensure ownership on several files including JENKINS_HOME. Therefore if High Availability is setup, this could lead to I/O Errors - for example when upgrading one node while another one is still running. More information is available in 
+link:https://issues.jenkins.io/browse/JENKINS-23273[JENKINS-23273].
+
+To workaround this problem, you can skip this script by adding the property
```
```suggestion
To workaround this problem, you can skip this script by adding the property
[source]
```

krisstern commented on content/doc/book/updating/other.adoc:
```diff
@@ -0,0 +1,39 @@
+== FreeBSD
+
+When Jenkins is installed from the FreeBSD `pkg` system, it should be upgraded with `pkg`.
+Connect to your FreeBSD machine, get root access and run the commands:
```
```suggestion
Connect to your FreeBSD machine, get root access and run the commands:
[source]
```

krisstern commented on content/doc/book/updating/other.adoc:
````diff
@@ -0,0 +1,39 @@
+```
+pkg update
+```
+```
````
````suggestion
[source]
```
````

krisstern commented on content/doc/book/updating/other.adoc:
````diff
@@ -0,0 +1,39 @@
+```
+pkg upgrade jenkins
+```
+```
````
````suggestion
[source]
```
````

krisstern commented on content/doc/book/updating/servlet-containers.adoc:
```diff
@@ -0,0 +1,41 @@
+Refer to the servlet container’s documentation for how to add startup arguments.
+
+== Tomcat
+Use environment variable CATALINA_OPTS to add:
```
```suggestion
Use environment variable CATALINA_OPTS to add:
[source]
```

krisstern commented on content/doc/book/updating/servlet-containers.adoc:
````diff
@@ -0,0 +1,41 @@
+```
+-Dorg.apache.tomcat.util.buf.UDecoder.ALLOW_ENCODED_SLASH=true.
+```
+```
````
````suggestion
[source]
```
````

krisstern commented:
Better be consistent about the styling when it comes to website development.

Vandit1604 commented:
I'll keep that in mind 
i totally forgot adding the [source] tag before code blocks 
thanks for the review

krisstern commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,108 @@
+service jenkins start
+```
+
+# Based on Operating Systems
````
```suggestion
#Based on Operating Systems
```

Vandit1604 replied:
This is a heading 
that's why we can't remove the space

krisstern replied:
I am not sure, but I don't think you are supposed to have any 0 level sections here

krisstern replied:
@Vandit1604 Are you sure you are supposed to use a # for the heading, I see you using the syntax `#heading` in a few instances in the same file above.

Vandit1604 commented:
I guess this commit will solve the failing checks

krisstern commented:
One more thing we will need to cater to: https://github.com/jenkins-infra/jenkins.io/issues/5059#issuecomment-1375659810

Basically for macOS you will need to provide at least a hyperlink to point to the external resource like the one Mark gave earlier.

krisstern commented:
And we have some broken image links on the Windows page

krisstern commented:
I am seeing this at the top of at least the Windows file: `:toc:left :toclevels: 4 :hide-uri-scheme: :imagesdir: ../../resources`.

Just to state again, it should be the following with a space in-between:
`:toc: left`

Vandit1604 commented:
Right now the TOC is populated with all the headings
but normally TOC should only contain level 1 and 2 heading which it does by default 
so removing the value from `:toclevels: value` can solve the problem of this big TOC that covers the page content
what are your thoughts @krisstern

kmartens27 commented on content/doc/book/updating/servlet-containers.adoc:
````diff
@@ -0,0 +1,43 @@
+```
+which is needed for Blue Ocean.
+
+It is recommended to configure it in the script `$CATALINA_BASE/bin/setenv.sh (Linux) or `%CATALINA_BASE%\bin\setenv.bat` (Microsoft Windows) that you’ll create to customize your application server.
````
```suggestion
It is recommended to configure it in the script you’ll create to customize your application server as `$CATALINA_BASE/bin/setenv.sh` (Linux) or `%CATALINA_BASE%\bin\setenv.bat` (Windows).
```
small update for message flow, if this removes the original intention, disregard

kmartens27 commented on content/doc/book/updating/servlet-containers.adoc:
```diff
@@ -0,0 +1,43 @@
+It is recommended to configure it in the script `$CATALINA_BASE/bin/setenv.sh (Linux) or `%CATALINA_BASE%\bin\setenv.bat` (Microsoft Windows) that you’ll create to customize your application server.
+
+== Custom container installations
+If you use a custom container, you will find the jenkins.war or jenkins-oc.war file in the deploy directory of your container. For example, `/usr/local/jboss/server/default/deploy/jenkins.war` would be the location for a default JBoss installation.
```
```suggestion
If you use a custom container, you will find the jenkins.war or jenkins-oc.war file in the deploy directory of your container.
For example, `/usr/local/jboss/server/default/deploy/jenkins.war` would be the location for a default JBoss installation.
```
one sentence per line

kmartens27 commented on content/doc/book/updating/war-file.adoc:
```diff
@@ -0,0 +1,43 @@
+== Replace the WAR file
+The Jenkins Web application ARchive (WAR) file can be started from the command line like this:
+
+. Download the 
+link:https://www.jenkins.io/download/[latest Jenkins WAR file]
+to an appropriate directory on your machine
```
```suggestion
. Download the link:https://www.jenkins.io/download/[latest Jenkins WAR file] to an appropriate directory on your machine.
```
one sentence per line

kmartens27 commented on content/doc/book/updating/war-file.adoc:
```diff
@@ -0,0 +1,43 @@
+link:https://www.jenkins.io/download/[latest Jenkins WAR file]
+to an appropriate directory on your machine
+
+. Open up a terminal/command prompt window to the download directory
```
```suggestion
. Open up a terminal/command prompt window to the download directory.
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+
+There are two ways to upgrade Jenkins in Windows.
+One is via Windows MSI installer
+Other one is discussed later on the page.
```
```suggestion
Other one is discussed later on the page.
```
Instead of saying one is discussed later, say what the name of it is here and link to it. This information does not help the user how it is written.

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+Other one is discussed later on the page.
+
+== Upgrading via MSI Installer
+you can download the latest LTS or Weekly MSI Installer to upgrade Jenkins.
```
```suggestion
You can download the latest LTS or Weekly MSI Installer to upgrade Jenkins.
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+== Upgrading via MSI Installer
+you can download the latest LTS or Weekly MSI Installer to upgrade Jenkins.
+
+Just Download the Jenkins 
+link:https://www.jenkins.io/download/#downloading-jenkins[MSI Installer] from here and install the latest
+Jenkins LTS or Weekly.
+for more information visit this page.
+link:https://www.jenkins.io/doc/book/installing/windows/#windows-msi-installers[Installing Jenkins On Windows via MSI Installer]
```
```suggestion
Just Download the Jenkins link:https://www.jenkins.io/download/#downloading-jenkins[MSI Installer] from here and install the latest Jenkins LTS or Weekly.
For more information refer to the link:https://www.jenkins.io/doc/book/installing/windows/#windows-msi-installers[Installing Jenkins On Windows via MSI Installer] page.
```
suggestion for formatting

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+for more information visit this page.
+link:https://www.jenkins.io/doc/book/installing/windows/#windows-msi-installers[Installing Jenkins On Windows via MSI Installer]
+
+=== 1. Stop the running Jenkins service
```
```suggestion
=== Stop the running Jenkins service
```
remove the numbered section headers

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+
+=== 1. Stop the running Jenkins service
+
+To stop Jenkins Please avoid shutting down the Java process or the Windows service. These are not usual commands. Use those only if your Jenkins is causing problems.
```
```suggestion
To stop Jenkins please avoid shutting down the Java process or the Windows service. These are not usual commands.
Use those only if your Jenkins is causing problems.
```
This section does not seem very clear, please add additional context/text to help clarify the statement

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+
+To stop Jenkins Please avoid shutting down the Java process or the Windows service. These are not usual commands. Use those only if your Jenkins is causing problems.
+
+use the following command to stop server
```
```suggestion
Use the following command to stop the server:
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+use the following command to stop server
+`http://[jenkins-server]/exit`
+
+Example: if my local PC is running Jenkins at port 8080, it will be
```
```suggestion
For example, if my local PC is running Jenkins at port 8080, it will be
```
small wording suggestion

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+Example: if my local PC is running Jenkins at port 8080, it will be
+`http://localhost:8080/exit`
+
+=== 2. Go to the location where your Jenkins instance is installed
```
```suggestion
=== Go to the location where your Jenkins instance is installed
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+
+=== 2. Go to the location where your Jenkins instance is installed
+	
+By Default Jenkins is installed here: 
```
```suggestion
By default, Jenkins is installed here: 
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+By Default Jenkins is installed here: 
+`C:\Program Files\Jenkins`
+
+=== 3. Rename the war folder to war.old
+This is important if the upgrade process fails 
+you won't lose the backup of your Jenkins instance
```
```suggestion
=== Rename the war folder to war.old

This is important if the upgrade process fails, as you won't lose the backup of your Jenkins instance
```
small suggestion for readability

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+This is important if the upgrade process fails 
+you won't lose the backup of your Jenkins instance
+
+=== 4. Rename jenkins.war file to jenkins.war.old
+This is important for backup so *Don't Replace and Don't Delete* the old war file.
```
```suggestion
=== Rename jenkins.war file to jenkins.war.old

This is important for the backup, so do not replace or delete the old war file.
```
small update, just to make it read smoother

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+=== 5. Copy the new jenkins.war you downloaded into your Jenkins installation directory
+This is the upgraded jenkins.war which will update your Jenkins Instance
+
+=== 6. Start Jenkins service
```
```suggestion
=== Start Jenkins
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,71 @@
+
+=== 6. Start Jenkins service
+
+To start the service you need to open the terminal in the directory where your Java WAR File is present and run the following command
```
```suggestion
To start the service, you need to open the terminal in the directory where your Java WAR File is present and run the following command:
```
punctuation suggestion

kmartens27 commented on content/doc/book/updating/windows.adoc:
````diff
@@ -0,0 +1,71 @@
+java -jar jenkins.war
+```
+
+=== 7. Observe the creation of a new war directory which will now contain the contents of the new jenkins.war archive
+
+By doing so, you can avoid losing your setup and builds that happen when a WAR file is replaced.
````
```suggestion
=== Observe the creation of a new war directory which will now contain the contents of the new jenkins.war archive

By doing so, you can avoid losing your setup and builds that happen when a WAR file is replaced.
```
Is this its own step or could it be added to the previous step as a result, or just a separate result that is not part of the steps?

kmartens27 reviewed (changes requested):
Thanks so much for your work on this @Vandit1604, I made some suggestions on formatting/context, but please let me know if you have any questions that I can help with!

Vandit1604 commented:
Thanks @kmartens27 for the suggestions 
I'll also update the structure of the pages like the LInux page

Vandit1604 commented:
there were so many suggestions 
@kmartens27 can you check if i addressed all of them correctly?

kmartens27 commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,65 @@
+[.boxshadow]
+image:updating/check-jenkins-version.png[alt="Checking Version of Updated Jenkins Instance",width=80%]
+
+The upgraded Jenkins version can be seen at the bottom right corner of the Jenkins UI.
```
```suggestion
The upgraded Jenkins version can be seen in the bottom right corner of the Jenkins UI.
```

kmartens27 commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,103 @@
+---
+
+Red Hat / CentOS ::
+Beginning in CloudBees Jenkins Platform version 2.319.1.5, the epel-release package is required to upgrade CloudBees Jenkins Platform using an RPM.
```
More of an info point, but CloudBees Jenkins Platform will be reaching [end of life](https://docs.cloudbees.com/docs/cloudbees-common/latest/maintenance-lifecycle#fixed-release) this coming summer (June 30, 2023) This may still be relevant but will not be supported in the future, fine to leave for now but just wanted to share that info!

kmartens27 replied:
I think it's fine to leave in for now, as it still does apply. Thanks again @Vandit1604

Vandit1604 replied:
@dduportal It does seems weird i'll change it to [ **Jenkins** ] just now

Vandit1604 replied:
I have removed the commercial name does everything else looks good to you now ?

dduportal replied:
Thanks a lot for this work!

dduportal replied:
I am not really sure, as I do not use CentOS at all, sorry. I would defer to @MarkEWaite or @krisstern if they know the reason for this addon (is there an intent or is is a copy and paste from cloudbees doc ?)

at first sight, the version number (with 4 digits) should be changed (as the 4 digits versioning pattern is cloudbees only while jenkins LTS only has 3 digits)

kmartens27 commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,69 @@
+
+== Upgrade steps on Windows
+
+There are two ways to upgrade Jenkins in Windows.
+One is via Windows MSI installer 
+Other one is by Replacing the 
+link:https://jenkins.io/doc/book/updating/windows/#replacing-the-war-file[WAR File].
```
```suggestion
There are two ways to upgrade Jenkins in Windows:
* Windows MSI installer 
or
* Replacing the  link:https://jenkins.io/doc/book/updating/windows/#replacing-the-war-file[WAR File].
```

kmartens27 commented on content/doc/book/updating/windows.adoc:
````diff
@@ -0,0 +1,69 @@
+java -jar jenkins.war
+```
+
+Observe the creation of a new war directory which will now contain the contents of the new jenkins.war archive. By doing so, you can avoid losing your setup and builds that happen when a WAR file is replaced.
````
```suggestion
Observe the creation of a new war directory which will now contain the contents of the new jenkins.war archive.
By doing so, you can avoid losing your setup and builds that happen when a WAR file is replaced.
```
one sentence per line suggestion

kmartens27 commented on content/doc/book/updating/_pre-upgrade-wizard.adoc:
```diff
@@ -0,0 +1,18 @@
+
+. Verify all running builds are complete.
+
+. Java 11 is required for your Jenkins instance, the operations center, Jenkins controllers, and all agents for Jenkins LTS 2.303.3.
```
```suggestion
. Java 11 is required for a modern Jenkins instance, the operations center, Jenkins controllers, and all agents for Jenkins as of the 2.357 weekly and Long Term Support release of 2.361.1.
```
[Java 11 is required](https://www.jenkins.io/blog/2022/06/28/require-java-11/#:~:text=Beginning%20with%20Jenkins%202.357%20(released,)%2C%20Jenkins%20supports%20Java%2017.) when updating to Jenkins LTS 2.361.1 (we are now on 2.375.1) or weekly (starting with 2.357, and we are now on 2.386)
Suggesting to make sure the current requirements are clear and visible

kmartens27 commented:
Hi @Vandit1604, thanks so much for taking care of the suggestions! I saw that there were still a couple from my previous review, so just curious to see if there were any questions on those, and I added a couple more that I found during my new review.

krisstern commented:
> Right now the TOC is populated with all the headings but normally TOC should only contain level 1 and 2 heading which it does by default as stated in [ASCIDOC docs](https://docs.asciidoctor.org/asciidoc/latest/toc/levels/#set-toclevels) so removing the value from `:toclevels: value` can solve the problem of this big TOC that covers the page content what are your thoughts @krisstern

Using `:toclevels: 3` as you are doing now for the PR looks alright to me. I don't really have a very strong opinion about this. Let me what other reviewer might have in mind.

Vandit1604 commented:
> Hi @Vandit1604, thanks so much for taking care of the suggestions! I saw that there were still a couple from my previous review, so just curious to see if there were any questions on those, and I added a couple more that I found during my new review.

@kmartens27 thanks for the reviews. I think i have addressed all of them now ,does the site look good now

Vandit1604 commented:
@MarkEWaite thanks for the commits everything looks good now
But i have a question how can I add the `Updating Jenkins` section in the **Documentation** button on navbar
see the attached image 
![Screenshot from 2023-01-10 13-41-09](https://user-images.githubusercontent.com/107131545/211496968-f700a89d-6ab6-4a43-ae79-47cf23fea523.png)

kmartens27 reviewed (approved):
Thanks so much @Vandit1604 for all of this, it looks great!
As for adding it to the navigation menu list, we are currently working on a different task related to the navigation/menu bar and the options displayed. As we continue to progress on that, this can be part of the discussion as we go.

MarkEWaite commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,102 @@
+#Stop running Jenkins service using the following command:
+
+sudo su -
+service jenkins stop
```
Jenkins installed with Linux installers uses `systemd`.  The `service` command Red Hat Enterprise Linux (and its derivatives) says that it runs System V init scripts.  It does not mention running `systemd` commands.  The Ubuntu 22.04 `service` man page says that it runs both System V init and `systemd` commands.  I think we should use `systemd` rather than using System V init.
```suggestion
systemctl stop jenkins
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,102 @@
+```
+#Take backup of Jenkins home directory. In my case my Jenkins home directory is /var/lib/jenkins.
+
+cd /var/lib
+tar -cvzf jenkins_date.tar.gz jenkins/
+mv jenkins_date.tar.gz $HOME
````
Replace with a link to the backup page, because there are many different ways to perform a backup and they depend on the environment.  For example, if running Ubuntu 22.04 with OpenZFS file system, the best backup may be to take file system snapshot, not a tar file.  Likewise with OpenSUSE 15.4 with btrfs.

Should not be in a code block because we should not describe all the backup alternatives here.  Link them to the correct page.
```suggestion
See the link:/doc/book/system-administration/backing-up/[backup instructions] for alternatives.
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,102 @@
+mv jenkins_date.tar.gz $HOME
+```
+
+3) Take The backup of the current Jenkins war file ::
+[source,bash]
+```
+# Take the backup of Jenkins current version binary using following commands:
+
+cd /usr/share/java/
+cp -p jenkins.war jenkins.war.old
+```
+
````
Not needed, either include it in the backup or download it again.
```suggestion
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,102 @@
+cp -p jenkins.war jenkins.war.old
+```
+
+4) Download The New LTS Jenkins Version ::
+[source,bash]
+```
+#Download the new version of Jenkins WAR file using the following commands:
+
+wget https://updates.jenkins-ci.org/latest/jenkins.war
+```
````
We **must not upgrade by copying in a new war file** if the previous installation was done with a `deb` or an `rpm`.  Upgrade must be performed by the same type of command as the initial install.  That is specific to the Linux distribution.
```suggestion
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,102 @@
+[source,bash]
+```
+#Start the Jenkins service using the following command:
+service jenkins start
````
Use `systemd` not System V init.
```suggestion
systemctl start jenkins
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,102 @@
+
+* Update Operations center with `yum upgrade jenkins-oc`
+
+RPM/YUM with High Availability (Important) ::
+
+The RPM package contains a post-install script to ensure ownership on several files including JENKINS_HOME.
+Therefore if High Availability is setup, this could lead to I/O Errors - for example when upgrading one node while another one is still running.
+More information is available in link:https://issues.jenkins.io/browse/JENKINS-23273[JENKINS-23273].
```
This issue was fixed 5 years ago.  No need to mention it.
```suggestion
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,102 @@
+---
+
+Red Hat / CentOS ::
+Beginning in Jenkins version 2.319, the epel-release package is required to upgrade Jenkins using an RPM.
+However, some CentOS distributions do not include it.
+For those CentOS distributions, run the following command before you begin the installation procedure: 
+`sudo yum install epel-release`.
```
No longer required due to the Jenkins transition from System V initi to systemd
```suggestion
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,102 @@
+Therefore if High Availability is setup, this could lead to I/O Errors - for example when upgrading one node while another one is still running.
+More information is available in link:https://issues.jenkins.io/browse/JENKINS-23273[JENKINS-23273].
+
+To workaround this problem, you can skip this script by adding the property
+[source]
+```
+JENKINS_INSTALL_SKIP_CHOWN="true"
+```
+under `/etc/sysconfig/jenkins`.
+
````
No longer needed
```suggestion
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,102 @@
+
+Debian / Ubuntu ::
+* Update Controller with `apt-get update && apt-get install jenkins`
+
+* Update Operations center with `apt-get update && apt-get install jenkins-oc`
```
Jenkins does not have anything called Operations Center.  That is a part of the CloudBees products.  We rely on CloudBees to document their products.
```suggestion
```

MarkEWaite commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,102 @@
+* Update Operations center with `apt-get update && apt-get install jenkins-oc`
+
+openSUSE / SUSE Linux ::
+* Update Controller with `zypper install cloudbees-core-cm`
+* Update Operations center with `zypper install cloudbees-core-oc`
```
The command on OpenSUSE is `update` rather than `install`
```suggestion
* Update Controller with `zypper update jenkins`
```

MarkEWaite commented on content/doc/book/updating/war-file.adoc:
```diff
@@ -0,0 +1,37 @@
+
+[unordered.stack]
+Replace the WAR file ::
+The Jenkins Web application ARchive (WAR) file can be started from the command line like this:
```
Seems like this line was describing something different, not an upgrade
```suggestion
```

MarkEWaite commented on content/doc/book/updating/windows.adoc:
```diff
@@ -0,0 +1,72 @@
+
+== Upgrade steps on Windows
+
+There are two ways to upgrade Jenkins in Windows:
```
Needs instructions to guide the user to upgrade with the same technique they used for installation.  If they installed with MSI, they should upgrade with MSI.  If they installed by copying the war file, they should upgrade by replacing the war file.

Vandit1604 replied:
Thanks for the suggestions @MarkEWaite I'll add the instructions

MarkEWaite reviewed (changes requested):
Thanks so much for your work on this.  It is a great start.  @Stackscribe and I reviewed this during docs office hours 20 Jan 2023.  Recording will be available if it helps.

Vandit1604 commented:
I have added the instructions now and I'll also have to add the Updating Jenkins section under the Documentation dropdown, i'll do that after this is good to go

krisstern commented:
> I have added the instructions now and I'll also have to add the Updating Jenkins section under the Documentation dropdown, i'll do that after this is good to go

@Vandit1604 It may be a better idea to wait and see if that is the consensus as mentioned previoulsy in https://github.com/jenkins-infra/jenkins.io/pull/5888#pullrequestreview-1242274052 by Kevin.

MarkEWaite reviewed (changes requested):
Reviewed in Docs Office Hours with Mark Waite, @StackScribe , and @dheerajodha

kmartens27 commented on content/doc/book/updating/linux.adoc:
````diff
@@ -0,0 +1,61 @@
+
+include::doc/book/updating/_pre-upgrade-wizard.adoc[]
+
+== Upgrade steps on Linux
+[unordered.stack]
+1) Stop your Jenkins Instance :: 
+[source,bash]
+```
+#Stop running Jenkins service using the following command:
+
+sudo su -
+systemctl stop jenkins
+```
````
== Upgrade steps on Linux
[[upgrade.linux]]
. Stop your Jenkins instance using the following command:

[source,bash]
```
sudo su -
systemctl stop jenkins
```

This looks to be another set of steps and should be using a numbered list format as well. I had to remove the suggestion formatting so that I could put this in properly. The [anchor](https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#add-additional-anchors-to-a-section) would also need to be updated accordingly, as the `.` delimiter is not interpreted correctly with just a single set of []

Vandit1604 replied:
Because of the same reason of numbers order start from 1 as long as we write any content between [ordered list](https://docs.asciidoctor.org/asciidoc/latest/lists/ordered/) we can't use ordered list formatting

kmartens27 replied:
I was able to find [this information](https://docs.antora.org/antora/latest/asciidoc/description-lists/#complex) and [this information](https://docs.asciidoctor.org/asciidoc/latest/lists/continuation/), which is what I had been thinking of when reviewing the lists. There are ways to keep things in line within a list, provided the correct formatting is used. However, I do see the difference between the ordered and description lists in this case. Would it be possible to try this out?

Vandit1604 replied:
Thanks @kmartens27 for your review and time.
I have changed the list style in linux.adoc and windows.adoc, whenever you get time please take a look

kmartens27 commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,70 @@
+////
+This file is only meant to be included as a snippet in other documents.
+There is a version of this file for the general 'Updating Jenkins' page (index.adoc).
+This file is for the index.adoc page used in the general 'Updating Jenkins' page.
```
This file is not included on the index page, so it does not appear at this time. It is included in other updating pages, but if it is general and should be in the index, it should be added there too.

Vandit1604 replied:
Actually i followed the [content/doc/book/installing/index.adoc](https://github.com/jenkins-infra/jenkins.io/blob/master/content/doc/book/installing/index.adoc) for the file structure that's why this line is there and it should be removed from there.

kmartens27 commented on content/doc/book/updating/_pre-upgrade-wizard.adoc:
```diff
@@ -0,0 +1,14 @@
+////
+This file is only meant to be included as a snippet in other documents.
+There is a version of this file for the general 'Updating Jenkins' page (index.adoc).
+This file is for the index.adoc page used in the general 'Updating Jenkins' page.
```
```suggestion
This file is for the index.adoc page used in the general 'Updating Jenkins' page.
```
This does not appear to be included in the index page, so needs to be added there as well (if that is the idea).

kmartens27 commented on content/doc/book/updating/linux.adoc:
```diff
@@ -0,0 +1,61 @@
+
+See the link:/doc/book/system-administration/backing-up/[backup instructions] for alternatives.
+
+5) Start The Jenkins Service ::
```
this goes from step 2 to step 5, I would advise using the [ordered list formatting](https://docs.asciidoctor.org/asciidoc/latest/lists/ordered/) so that the numbers are kept in order by the document itself.

Vandit1604 replied:
I have used the description list after experimenting with [ordered list formatting](https://docs.asciidoctor.org/asciidoc/latest/lists/ordered/) but the problem is ordered list formatting won't keep the order by itself if not written like this

```
. Protons
. Electrons
. Neutrons
```
writing some paragraphs between ordered list formatting won't work, we can make do use _start_ but it isn't recommended, [see here](https://docs.asciidoctor.org/asciidoc/latest/lists/ordered/)

kmartens27 reviewed (changes requested):
Hi @Vandit1604, thanks again for all your continued work on this. I was doing another review and found some areas that could use a bit of updating/modification. I have also included some asciidoc references for things such as anchors and ordered lists, as I believe some of the instructions could be updated with these features.

Let me know if you have any questions on any of my suggestions!

Vandit1604 commented:
Are there any further actions required to complete this PR?

MarkEWaite commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,66 @@
+
+The above image shows that there is no need to update any plugins.
+
+After upgrading Jenkins, you may encounter data compatibility issues with the new version.
+To ensure that our data is in the correct format, we should update it to the correct format.
+To check the data version update, follow these steps:
```
I think this should be a link to the existing page that describes the "Manage old data" choices and reasons.
```suggestion
Sometimes a Jenkins plugin changes the data that it is storing.
In other cases, a plugin that has been uninstalled may have left data in the system that Jenkins no longer needs.
The link:/doc/book/managing/plugins/#removing-old-data["Manage old data"] page allows administrators to remove those unused items.
```

MarkEWaite commented on content/doc/book/updating/_update-wizard.adoc:
```diff
@@ -0,0 +1,66 @@
+To check the data version update, follow these steps:
+
+. Go to the Jenkins dashboard
+. Select *Manage Jenkins* in the navigation pane, on the left side of the Jenkins dsahboard.
```
Spelling fix
```suggestion
. Select *Manage Jenkins* in the navigation pane, on the left side of the Jenkins dashboard.
```

MarkEWaite commented on content/doc/book/updating/index.adoc:
```diff
@@ -0,0 +1,20 @@
+Jenkins can be installed with operating system specific installers like link:https://www.jenkins.io/doc/book/installing/windows/#windows-msi-installers[MSI], link:https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos[RPM], and link:https://www.jenkins.io/doc/book/installing/linux/#debianubuntu[DEB] installers.
+When upgrading Jenkins, the process used must match the original installation method:
+
+* If Jenkins was installed with an operating system specific installer, it should be upgraded with an operating system specific installer
```
Windows and Linux installers behave differently in this case
```suggestion
* If Jenkins was installed with an operating system specific Linux installer, it should be upgraded with the same operating system specific Linux installer
* If Jenkins was installed with the Windows installer, it should be upgraded using the upgrade selection inside Jenkins
```

MarkEWaite reviewed (changes requested):
Several items that need to be updated, then another review of the document is needed.

Vandit1604 commented:
Given the current level of traction on this PR, I believe it may be best to close it for now. I am happy to revisit and continue working on it in the future should the need arise.
